pep_sphinx_extensions/tests/pep_zero_generator/test_parser.py

```diff
@@ -0,0 +1,84 @@
+from pathlib import Path
+
+import pytest
+
+from pep_sphinx_extensions.pep_zero_generator import parser
+from pep_sphinx_extensions.pep_zero_generator.author import Author
+from pep_sphinx_extensions.pep_zero_generator.errors import PEPError
+from pep_sphinx_extensions.tests.utils import AUTHORS_OVERRIDES
+
+
+def test_pep_repr():
+    pep8 = parser.PEP(Path("pep-0008.txt"), AUTHORS_OVERRIDES)
+
+    assert repr(pep8) == "<PEP 0008 - Style Guide for Python Code>"
+
+
+def test_pep_less_than():
+    pep8 = parser.PEP(Path("pep-0008.txt"), AUTHORS_OVERRIDES)
+    pep3333 = parser.PEP(Path("pep-3333.txt"), AUTHORS_OVERRIDES)
+
+    assert pep8 < pep3333
+
+
+def test_pep_equal():
+    pep_a = parser.PEP(Path("pep-0008.txt"), AUTHORS_OVERRIDES)
+    pep_b = parser.PEP(Path("pep-0008.txt"), AUTHORS_OVERRIDES)
+
+    assert pep_a == pep_b
+
+
+def test_pep_details():
+    pep8 = parser.PEP(Path("pep-0008.txt"), AUTHORS_OVERRIDES)
+
+    assert pep8.details(title_length=80) == {
+        "authors": "GvR, Warsaw, Coghlan",
+        "number": 8,
+        "status": " ",
+        "title": "Style Guide for Python Code",
+        "type": "P",
+    }
+
+
+def test_pep_details_short_title():
+    pep8 = parser.PEP(Path("pep-0008.txt"), AUTHORS_OVERRIDES)
+
+    assert pep8.details(title_length=10) == {
+        "authors": "GvR, Warsaw, Coghlan",
+        "number": 8,
+        "status": " ",
+        "title": "Style ...",
+        "type": "P",
+    }
+
+
+@pytest.mark.parametrize(
+    "test_input, expected",
+    [
+        (
+            "First Last <user@example.com>",
+            [Author(last_first="Last, First", nick="Last", email="user@example.com")],
+        ),
+        (
+            "First Last",
+            [Author(last_first="Last, First", nick="Last", email="")],
```

Comment:
Shouldn't we also test the legacy `user@example.com (First Last)` format here, as well as the manually pseudo-obfuscated variants?

Reply:
Yup, LGTM thanks